ibis/backends/snowflake/__init__.py

```diff
@@ -239,7 +251,54 @@ def normalize_name(name):
                 return name
 
         self.con.dialect.normalize_name = normalize_name
-        return res
+        self.database_name = self._default_database = dbparams["database"]
+        self._default_schema = dbparams["schema"]
+        return eng
+
+    @classmethod
+    def from_snowpark(cls, session) -> Backend:
+        """Create an Ibis Snowflake backend from a Snowpark session.
+
+        Parameters
+        ----------
+        session
+            A snowpark session.
+
+        Returns
+        -------
+        Backend
+            An Ibis Snowflake backend instance.
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> import snowflake.snowpark as sp  # doctest: +SKIP
+        >>> session = sp.Session.builder.configs(...).create()  # doctest: +SKIP
+        >>> con = ibis.snowflake.from_snowpark(session)  # doctest: +SKIP
+        >>> batting = con.tables.BATTING  # doctest: +SKIP
+        >>> batting[["playerID", "RBI"]].head()  # doctest: +SKIP
+        ┏━━━━━━━━━━━┳━━━━━━━┓
+        ┃ playerID  ┃ RBI   ┃
+        ┡━━━━━━━━━━━╇━━━━━━━┩
+        │ string    │ int64 │
+        ├───────────┼───────┤
+        │ abercda01 │     0 │
+        │ addybo01  │    13 │
+        │ allisar01 │    19 │
+        │ allisdo01 │    27 │
+        │ ansonca01 │    16 │
+        └───────────┴───────┘
+        """
+        con = session._conn._conn
+        # workaround a misconfigured implementation of the DBAPI cursor
+        # when con._paramstyle is not set to "pyformat"
+        #
+        # when using snowpark, con._paramstyle is set to qmark
+        con._paramstyle = "pyformat"
```

Comment:
Does mutating the internals of the snow park session connection break anything if the user tries to reuse the snow park session for snow park stuff later (not sure if that workflow is possible/likely)?

Reply:
Good question, I'm not 100% sure. I'll see if I can add a test that connects to ibis and triggers use of parameters with snowpark.

Reply:
This does in fact break any further use of bind parameters in snowflake.

I can look into whether there's a way around this.

Reply:
Ok, addressed this.